identity_iota_core/src/rebased/client/read_only.rs

```diff
@@ -190,9 +194,9 @@ impl IdentityClientReadOnly {
 
     // use first non-None value as result
     let mut identity_outcome: Option<Identity> = None;
-    while let Some(result) = all_futures.try_next().await? {
-      if result.is_some() {
-        identity_outcome = result;
+    while let Some(result) = all_futures.next().await {
```

Comment:
If you like `stream`'s API you could try:
```rust
// spawn all checks
let all_futures = FuturesUnordered::<Pin<Box<dyn Future<Output = Result<Option<Identity>, Error>> + Send>>>::new();
all_futures.push(Box::pin(resolve_new(self, object_id)));
all_futures.push(Box::pin(resolve_migrated(self, object_id)));
all_futures.push(Box::pin(resolve_unmigrated(self, object_id)));

all_futures
   .filter_map(|res| Box::pin(async move { res.ok().flatten() }))
   .next()
   .await
   .ok_or_else(|| Error::DIDResolutionError(format!("could not find DID document for {object_id}")))
```

Reply:
That's way more beautiful and less clunky. Applied! ^^